src/field/berry.c

```diff
@@ -1099,91 +1126,96 @@ void ClearBerryTrees(void)
     struct SaveBlock1 *saveBlock1 = &gSaveBlock1;
     struct BerryTree berryTree = gBlankBerryTree;
 
-    for (i = 0; i < (u8)ARRAY_COUNT(saveBlock1->berryTrees); i++) // casting to u8 fixes a mismatched signed compare. what
+    for (i = 0; i < MAX_BERRY_TREES; i++)
         saveBlock1->berryTrees[i] = berryTree;
 }
 
-bool32 BerryTreeGrow(struct BerryTree *tree)
+// when the player does not interact with the tree for a period of time, this is called to advance the grow state.
+static bool32 BerryTreeGrow(struct BerryTree *tree)
 {
-    if (tree->growthSparkle)
+    if (tree->growthSparkle != FALSE)
         return FALSE;
     switch (tree->stage)
     {
-    case 0:
+    case BERRY_STAGE_NO_BERRY:
         return FALSE;
-    case 4:
+    case BERRY_STAGE_FLOWERING:
         tree->berryYield = CalcBerryYield(tree);
-    case 1:
-    case 2:
-    case 3:
+    case BERRY_STAGE_PLANTED:
+    case BERRY_STAGE_SPROUTED:
+    case BERRY_STAGE_TALLER:
         tree->stage++;
         break;
-    case 5:
+    case BERRY_STAGE_BERRIES:
         tree->watered1 = 0;
         tree->watered2 = 0;
         tree->watered3 = 0;
         tree->watered4 = 0;
         tree->berryYield = 0;
-        tree->stage = 2;
-        if (++tree->regrowthCount == 10)
+        tree->stage = BERRY_STAGE_SPROUTED;
+        if (++tree->regrowthCount == BERRY_REGROW_LIMIT)
             *tree = gBlankBerryTree;
         break;
     }
     return TRUE;
 }
 
-void BerryTreeTimeUpdate(s32 minutes)
+// totalMinutes is how long its been since the last berry tree update.
+void BerryTreeTimeUpdate(s32 totalMinutes)
 {
     int i;
     struct BerryTree *tree;
 
-    for (i = 0; i < (u8)ARRAY_COUNT(gSaveBlock1.berryTrees); i++)
+    for (i = 0; i < MAX_BERRY_TREES; i++)
     {
         tree = &gSaveBlock1.berryTrees[i];
 
-        if (tree->berry && tree->stage && !tree->growthSparkle)
+        if (tree->berry != BERRY_NONE && tree->stage != BERRY_STAGE_NO_BERRY && tree->growthSparkle == FALSE)
         {
-            if (minutes >= GetStageDurationByBerryType(tree->berry) * 71)
+            // the player has waited too long to water the berry. Reset the tree. This is because if the berry state is not in the unwatered state, the tree will grow anyway despite this check, which means BerryTreeGrow will handle the regrow process for this, removing the need for this check. This only handles the unwatered soil state.
+            if (totalMinutes >= GetStageDurationByBerryType(tree->berry) * 71)
             {
                 *tree = gBlankBerryTree;
             }
             else
             {
-                s32 time = minutes;
+                // not sure why Game Freak thought a temporary variable was necessary. because time is altered below, perhaps they thought it was unsafe to change it, even though that is not how passed arguments behave.
```

Comment:
Unhelpful comment

Reply:
i cant get rid of this temp var, so evidently GF put it there for some reason. I don't think I need vague things unexplained.

Reply:
This comment doesn't help anything. Please remove it.